rmf_websocket/test/test_client.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+#define CATCH_CONFIG_MAIN
+#include <rmf_utils/catch.hpp>
+
+#include <atomic>
+#include <thread>
+#include <websocketpp/config/asio.hpp>
+#include <websocketpp/server.hpp>
+#include <rmf_websocket/BroadcastClient.hpp>
+#include <string>
+#include <rclcpp/node.hpp>
+
+typedef websocketpp::server<websocketpp::config::asio> server;
+typedef server::message_ptr message_ptr;
+
+using websocketpp::lib::placeholders::_1;
+using websocketpp::lib::placeholders::_2;
+using websocketpp::lib::bind;
+
+
+std::atomic_bool terminate_server = false;
+std::atomic<int> num_msgs = 0;
+std::vector<nlohmann::json> msgs;
+// Define a handler for incoming messages
+void on_message(server* s, websocketpp::connection_hdl hdl, message_ptr msg)
+{
+  terminate_server = true;
+  num_msgs++;
+  msgs.push_back(nlohmann::json::parse(msg->get_payload()));
+}
+
+
+void run_server()
+{
+  server echo_server;
+
+  echo_server.set_access_channels(websocketpp::log::alevel::all);
+  echo_server.clear_access_channels(websocketpp::log::alevel::frame_payload);
+  echo_server.init_asio();
+  echo_server.set_reuse_addr(true);
+  // Set on_message handler
+  echo_server.set_message_handler(
+    bind(&on_message, &echo_server, ::_1, ::_2));
+
+
+  // Set the port number
+  echo_server.listen(9000);
```

Comment:
See if you can use an arbitrary port (usually done by using port 0, then checking the socket to see which port the OS picked) to avoid port conflicts problems, but it's not an issue if it's hard to do so in websocketpp.